Add documentation for scrapers

hancush opened the pull request:
## Description

Hark! A documentation site!

Succeeding build here: https://github.com/Metro-Records/scrapers-lametro/actions/runs/10622927449/job/29448271525

### Testing

- Review the documentation at https://metro-records.github.io/scrapers-lametro/ and confirm it's a good first draft.
- Try to run the Quarto docs locally following the instructions in the README and confirm you can preview the built site and make updates.

hancush committed:
Add scraper documentation

hancush committed:
Don't ignore _site/

hancush committed:
Add workflow to publish docs

hancush committed:
docker-compose -> docker compose

hancush committed:
Trigger build from PR

hancush committed:
Trigger build from PR, redux

hancush committed:
Trigger build from PR, redux #2

hancush committed:
Trigger build from PR, redux #3

hancush committed:
Remove PR trigger

hancush commented on .github/workflows/publish_docs.yml:
```diff
@@ -0,0 +1,34 @@
+          files: |
+            docs/*
+
+      - name: Set up Quarto
+        if: steps.docs-changed.outputs.any_changed == 'true'
+        uses: quarto-dev/quarto-actions/setup@v2
+
+      - name: Render and Publish
+        if: steps.docs-changed.outputs.any_changed == 'true'
+        uses: quarto-dev/quarto-actions/publish@v2
+        with:
+          path: docs
+          target: gh-pages
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
These steps are only run if there are changes in the `docs/` directory.

hancush committed:
Update README

hancush committed:
One more README update

hancush commented on docker-compose.councilmatic.yml:
```diff
@@ -0,0 +1,26 @@
+version: '2.4'
+
+services:
+  scrapers:
+    image: scrapers-lametro
+    container_name: scrapers-lametro
+    build: .
+    stdin_open: true
+    tty: true
+    volumes:
+      - .:/app
+    environment:
+      # Populate the local Councilmatic database
+      DATABASE_URL: postgresql://postgres:postgres@postgres:5432/lametro
+      DJANGO_SETTINGS_MODULE: pupa.settings
+      OCD_DIVISION_CSV: "/app/lametro/divisions.csv"
+    command: pupa update lametro
+    # Connect the scraper container to the app network
+    networks:
+      - app_net
+
+networks:
+  # Define connection to the app's Docker network
+  app_net:
+    name: la-metro-councilmatic_default
+    external: true
```
Cc @xmedr

xmedr replied:
ah this makes things way easier, ty!

hancush commented on docs/debugging.qmd:
```diff
@@ -0,0 +1,224 @@
+---
```
Migrated from the Councilmatic wiki.

hancush commented on docs/local-development.qmd:
````diff
@@ -0,0 +1,134 @@
+docker compose -f docker-compose.councilmatic.yml run --rm scrapers
+```
+
+### Useful pupa commands
````
Adapted from the Councilmatic wiki.

xmedr commented on docs/debugging.qmd:
```diff
@@ -0,0 +1,224 @@
+
+#### Legistar contains the wrong data
+
+Data issues can occur when the Legistar API or web interface displays the wrong information. This generally happens with Metro staff enters information that is incorrect.
```
might be a small typo. was it supposed to be something like:
> This generally happens \<when\> Metro staff enters information that is incorrect.

xmedr commented on docs/debugging.qmd:
```diff
@@ -0,0 +1,224 @@
+
+The dashboard lives on GitHub, [here](https://github.com/Metro-Records/la-metro-dashboard).
+
+Apache maintains thorough documentation of [core concepts](http://airflow.apache.org/docs/stable/concepts.html), as well as [navigating the UI](http://airflow.apache.org/docs/stable/ui.html). If you've never used Airflow before, these are great resources to start with!
```
this is super helpful!

xmedr commented on docs/debugging.qmd:
```diff
@@ -0,0 +1,224 @@
+
+![windowed_bill_scraping DAG tree view](https://i.imgur.com/9NDdEpy.png)
+
+Note that the scraping DAGs employ [branch operators](https://airflow.apache.org/docs/stable/concepts.html?highlight=branch#branching) to determine what kind of scraping task to run. Be sure to look closely to verify that the task you're expecting is green (succeeded), not pink (skipped).
```
this link leads to the core concepts page without highlighting anything. would this link make sense?

https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#concepts-branching

xmedr commented on docs/scrapers/events.qmd:
```diff
@@ -0,0 +1,22 @@
+
+## Key issues
+
+- Metro streams audio in both English and Spanish. They cannot associated multiple
+broadcast link with one event in Legistar, so they create two nearly identical events that
```
super small typo. is it supposed to be:
> They cannot associate multiple broadcast links with one event in Legistar, ...

xmedr reviewed (changes requested):
This is really great! And the quarto docs run/update locally. Got some little comments below.

I also wanted to ask: I found that it took some time to figure out how some terms in Legistar mapped to Metro. For example, that "matter" equates to "board report", or a "body" refers to a "legislation" (I might be wrong about that). Would it be useful to have a small note of alternative names that certain items might be called in the individual scrapers sections?

antidipyramid commented on docs/debugging.qmd:
```diff
@@ -0,0 +1,224 @@
+
+If scrapes are running, by far the most common source of issues is the scraper failing to capture changes to a bill or event. The root issue is that we rely on timestamps that should indicate an update to determine which bills and events to scrape. In reality, these timestamps do not always update when a change is made to an event or bill in Legistar. We have a couple of strategies to get around  ...
+
+1. During [the Friday support window](https://github.com/datamade/la-metro-dashboard/blob/ac416e5e03f6a97fb9b0c6112093c679cefb0d1c/dags/constants.py#L41-L59), we scrape all events and bills at the top of every hour.
```
Should we remove references to the Friday support window?

hancush replied:
Yes! Done.

antidipyramid reviewed (commented):
Amazing @hancush! Just one question inline.

hancush committed:
Add synonyms and core concepts, fix typos, strike Friday support reference

hancush commented:
@xmedr, @antidipyramid Thanks for the review, you two! Addressed your comments and published the changes manually: https://metro-records.github.io/scrapers-lametro/

Xavier, thanks for flagging synonyms! I've added those to each of the scraper pages, and I also added core concepts to the Jurisdiction scraper, since I found that really confusing when I got started. I also added links to the Open Civic Data documentation where relevant. They're really useful for understanding why data is structured the way it is.

Let me know if this looks good to merge!

xmedr reviewed (approved):
These are hugely helpful, thank you! Looks good to me
